### ISSUE TITLE
Resource Limits for Jaeger allInOne

### ISSUE THREAD

**Issue by clyang82** [jaegertracing/jaeger-operator#562]
I saw the example of all-in-one, there has defined `max-traces: 100000`. what is proposed memory limits for allInOne? considering the memory occupation of 100000 traces, jaeger query and jaeger collector, etc. Thanks.

```
apiVersion: jaegertracing.io/v1
kind: "Jaeger"
metadata:
  name: "my-jaeger"
spec:
  strategy: allInOne
  allInOne:
    image: jaegertracing/all-in-one:1.13
    options:
      log-level: debug
      query:
        base-path: /jaeger
  ui:
    options:
      dependencies:
        menuEnabled: false
      tracking: 
        gaID: UA-000000-2
      menu:
        - label: "About Jaeger"
          items:
            - label: "Documentation"
              url: "https://www.jaegertracing.io/docs/latest"
  storage:
    options:
      memory:
max-traces: 100000
```

/cc @jpkrohling @objectiser I did not find such data after searched the document and issues. Do you have such data? Thanks.

**Comment by objectiser** [comment]
The max traces were added to prevent an out of memory issue when being used in Istio - originally the memory storage had no limit as was only used for dev/testing purposes.

We haven't done any specific calculations to determine the figure currently - as it depends on the amount of information stored in each span (e.g. tags, logs, etc).

**Comment by objectiser** [comment]
@clyang82 Actually it may be worth considering using the badger local storage, which automatically manages the memory based on amount available I believe.

**Comment by clyang82** [comment]
@objectiser Thanks for your information. What do you mean `manages the memory based on amount available`? from this document - https://www.jaegertracing.io/docs/1.12/deployment/#badger-local-storage, I think it should consume small memory which is used to transfer spans from memory to local storage.

**Comment by objectiser** [comment]
@clyang82 Thought it also cached data in memory - but possibly wrong.

@burmanm Should be able to confirm.

**Comment by burmanm** [comment]
Badger uses memory mapping to access the files on the filesystem and as such the memory management of those is handled by the OS actually. When the kernel requires memory to other use, it will release the memory from badger to somewhere else and if it has available memory it will allow badger to cache more in the memory. 

Badger does require some memory in normal use obviously and the amount depends on the write pattern. This is because it needs to compact the data (remove stale data, rearrange data for easier reads, flush memtables to the disk etc). 

If you store the data to the tmpfs (ephemeral data) - then your OS settings play a large role in the memory consumption. All the data is stored in the memory, but they can be swapped to the disk by the kernel if necessary (depending on your kernel settings how aggressively this is done).

In short, you don't need a lot available memory to run badger storage backend - but more you have, better performance you will get.

**Comment by clyang82** [comment]
@burmanm Thanks for your elaboration. Do you have recommended memory value for 50000 traces? Does 500MB memory work?

**Comment by burmanm** [comment]
Do you mean using tmpfs or stored in the disk? 50k traces eats more than 500MB of disk space at least with the traces I've tried.

**Comment by clyang82** [comment]
No. I mean local storage device such as a disk, partition or directory.

**Comment by burmanm** [comment]
512MB should be enough (based on my quick testing). There are some options in the badger that could be used to reduce the memory requirements, but they're not exposed to the Jaeger at this point. I think I'll do something about that later..

**Comment by jpkrohling** [comment]
I'm closing this, as @burmanm has answered the question already.

**Comment by clyang82** [comment]
Yes. I should close this issue. Thanks @jpkrohling
